app/models/update_log.rb

```diff
@@ -4,14 +4,31 @@ def self.setting_record
     Setting.find_or_create_by(key: 'metrics_update')
   end
 
-  def self.log_update(time)
+  def self.last_update
+    setting = self.setting_record
+    setting.value.values.last
+
+  end
+
+  def self.log_updates(time)
     setting = self.setting_record
-    setting.value["last_update"] = time
+    last_update = setting.value.keys.last
+    if last_update === nil
+      new_update = 0
+    elsif last_update >= 10
+      setting.value.delete(last_update - 10)
+      new_update = last_update + 1
+    else 
+      new_update = last_update + 1
+    end
+    setting.value[new_update] = time
```

Comment:
I've I'm reading this right, this data structure is something like
```
{
  1 => time of update 1,
  2 => time of update 2,
 ...
}

And the numbers will just keep increasing, with the older values being removed?

I think that will work fine. It'll be more flexible and resilient to future changes if you store data about the update as another hash, instead of just the time value. For example, the value for each update number key could be something like `{ start_time: time, end_time: time }`. That way, as we add more things to the update logs (eg, keeping track of how many programs were active at the time of the update), we'll still be able to access the data the same way.